### PR TITLE
Add Expect support

### PR DESCRIPTION
**Pull request by nemeth**

### CODE REVIEW

File: examples/aws/spec/default/default.tfspec [hunk 1]
```diff
@@ -54,10 +54,21 @@ assert "module.ec2-instance.aws_instance" "this[0]" {
...
+expect "aws_s3_bucket" "b" {
+  bucket = "my-tf-test-bucket"
+  return {
+    id = "arn:aws:s3::0000000000:my-tf-test-bucket"
+  }
+}
+
+assert "aws_s3_bucket" "b" {
+    id = "arn:aws:s3::0000000000:my-tf-test-bucket"
+}
```
Comment (rcsehil) [inline examples/aws/spec/default/default.tfspec]: What is the difference between assert and expect?

File: README.md [hunk 2]
```diff
@@ -69,6 +69,16 @@ mock "aws_vpcs" "selected" {
...
 
+### Mock return values of resources
+
+Terraform code :
+```
+data "aws_vpcs" "selected" {
+  tags = {
+    service = "secure"
+  }
+}
```
Comment (rcsehil) [inline README.md]: This makes me confused, the title says resources but the example shows a single data resource and no mocking

File: lib/spec.go [hunk 3]
```diff
@@ -36,6 +38,15 @@ type Assert struct {
...
 }
 
+type Expect struct {
+	Type  string
+	Name  string
+	Query cty.Value
+	Data  cty.Value
+	Body  []byte
+	calls int
+}
```
Comment (rcsehil) [inline lib/spec.go]: What are Query, Data, Body and calls fields for?

File: examples/aws/spec/default/default.tfspec (hunk 1 above)
Comment (nemeth) [reply examples/aws/spec/default/default.tfspec]: a **mock** substitutes attributes in a data sources
an **expect** substitutes attributes in a resources, it mocks the state of a resource
an **assert** checks the values of attributes of resources or outputs as a result of terraform plan

File: lib/providers.go [hunk 4]
```diff
@@ -37,20 +38,20 @@ func (m *MockDataSourceReader) SetMock(mocks []*Mock) {
...
 // ReadDataSource returns a mock response for the datasource call
-func (m *MockDataSourceReader) ReadDataSource(config cty.Value) cty.Value {
-	var mockedResult cty.Value = config
+func (m *MockDataSourceReader) ReadDataSource(config cty.Value) (mockedResult cty.Value) {
+	mockedResult = config
 	for _, mock := range m.mockDataSources {
 		if mock.Query.RawEquals(config) {
 			mockedResult = mock.Call()
-			return mockedResult
+			return
```
Comment (tomelliot16) [inline lib/providers.go:46]: Not sure what's going on here

Comment (tomelliot16) [reply lib/providers.go:46]: Is this a golangism?

Comment (tomelliot16) [reply lib/providers.go:46]: Oh got it. Lol not sure I like it

**Review by tomelliot16 — APPROVED** [review]

**Pull request merged**